code/modules/paperwork/stamps.dm

```diff
@@ -12,6 +12,56 @@
 	item_color = "cargo"
 	pressure_resistance = 2
 	attack_verb = list("stamped")
+	var/stamp_color = "#264715"		//used for stamp marks, based on the item sprites.. not the on paper images(those are very bland)
+	var/stamp_description = "..."
+
+/obj/item/stamp/afterattack(atom/target, mob/user, proximity, params)
+	. = ..()
+	if(!ishuman(target) || !ishuman(user) || !proximity)
+		return
+
+	var/mob/living/carbon/human/H = target
+	var/attackedSide				//the stamp mark will appear on this side, NORTH = forwards, SOUTH = back, EAST = right, WEST = left
+	var/list/paramslist = params2list(params)
+	var/xOffset = text2num(paramslist["icon-x"]) - 16
+	var/yOffset = text2num(paramslist["icon-y"]) - 16
+
+	switch(H.dir)					//This makes the attackSide be the side that currently faces the 'camera'
+		if(NORTH)
+			attackedSide = SOUTH
+		if(SOUTH)
+			attackedSide = NORTH
+		if(EAST)
+			attackedSide = EAST
+		if(WEST)
+			attackedSide = WEST
+
+	var/icon/new_stamp_mark = icon('icons/effects/stamp_marks.dmi', "stamp[rand(1,3)]_[attackedSide]")
```

Comment:
I assume you've considered that already, and decided against it, but.. Do you maybe want to have three stamp icons (`stamp1`, `stamp2`, `stamp3`) in the dmi, each with several directions, and have this line be
```dm
var/icon/new_stamp_mark = icon('icons/effects/stamp_marks.dmi', "stamp[rand(1,3)]", attackedSide)
```
Or is this kind of directionality exactly what you're trying to avoid here (and treat every icon as if it's SOUTH direction)?

Reply:
currently I have four states for each varient of the mark, one for each direction. The other directions are empty for the state of the direction its defined as. bizzarely having a single image defined in one orientation and changing the direction in the definition of the image doesn't work, and results in the mark being visible regardless of the dir of the human its on. The only other alternative I know of would be having the sprite in all directions in one state, and cutting out the sprites for the directions it doesnt face, but that seems too costly for just avoiding +3 states in the dmi.

Reply:
I see. Makes sense (kinda). I could reproduce this.
And I could work around this by doing something like:

```dm
// Don't even need to specify the direction, if you have them as unidirectional in the .dmi
var/icon/new_stamp_mark_base = icon('icons/effects/stamp_marks.dmi', "stamp[rand(1,3)]", dir=SOUTH) 
// This is an empty four-directional icon.
var/icon/new_stamp_mark = new /icon('icons/effects/stamp_marks.dmi', "stamp1_e")
new_stamp_mark.Insert(new_stamp_mark_base, dir=H.dir)
```
The need for empty icon is unfortunate though.. I tried to create that one on the fly via either `new /icon()` or even
```dm
var/icon/new_stamp_mark = new /icon()
for(dir in GLOB.alldirs)
  new_stamp_mark.Insert(new /icon(), dir=dir)
```
and that even *almost* worked, except when the `H.dir==SOUTH` - that makes the stamp mark somehow visible from all directions (but stamping from non-SOUTH makes it work as expected!)

Either way, I suggest at least clarifying this quirk in the comment (and the fact that each stampX_Y only has Y direction in it).
And if you do end up going the "create empty icon" path, i guess that empty icon should be somewhere near the top, like  `icons/empty.dmi` or `icons/misc/empty.dmi`

Reply:
As ideal was this would be I cannot get it to work, unless the approach I have now is explicitly disallowed I am content with the code as is. I have clarified the way it currently works in code comments.